_releases/2022-07-06-1.5.0-released.md

```diff
@@ -0,0 +1,117 @@
+---
+title: Crystal 1.5.0 is released!
+version: 1.5.0
+summary:
+thumbnail: +
+author:
+---
+
+We are delivering a new release with several bugfixes and improvements. Below we list the most important or interesting changes, without mentioning several bugfixes and smaller enhancements. For more details, visit the [changelog](https://github.com/crystal-lang/crystal/releases/tag/1.5.0). Breaking changes are marked with ⚠️.
+
+Pre-built packages are available on [GitHub Releases](https://github.com/crystal-lang/crystal/releases/tag/1.5.0) and our official distribution channels.
+See [crystal-lang.org/install](https://crystal-lang.org/install/) for installation instructions.
+
+## Stats
+
+In this release we included [XXX changes since the 1.4.1 release](https://github.com/crystal-lang/crystal/pulls?q=is%3Apr+milestone%3A1.5.0) by XX contributors. We thank all the effort put into improving the language! ❤️
+
+## Remarkable changes
+
+### Method restrictions from instance variables
+
+When an instance variable is assigned the value of an untyped method argument, then the argument is restricted to share the same type as the instance variable.
+
+For instance, consider this code:
+
+```crystal
+class Foo
+  @x : Int64
+
+  def initialize(x)
+    @x = x
+  end
+end
+````
+
+Up to 1.4.1, `x` in `initialize` is unrestricted. This had several issues:
+
+ 1. If the user passes an incorrect argument, like `Foo.new('a')`, then the error wrongly blames `@x` for not having the right type.
+
+ 2. No autocast is performed, for instance, if we pass an `Int32` instead: `Foo.new(1)` fails.
+
+ 3. The generated documentation doesn't provide a hint of the type of the parameter `x`.
+
+From 1.5.0, in an assignment like `@x = x`, parameter `x` gets the type of `@x`, effectively solving the three issues above. Details can be read from [#12103](https://github.com/crystal-lang/crystal/pull/12103).
+
+> Note: if for some reason this new feature is bringing you a headache, let us know in the [issue tracker](https://github.com/crystal-lang/crystal/issues), and disable it passing `-Dno_restrictions_augmenter` in the build options.
+
+### Annotations allowed on method arguments
+
+It is now possible to add an annotation to a parameter of a method or macro, useful when needing to handle transformations in parameters. For instance, the following fictitious code allows to multiply the value received by the entrypoint `/multiply/{num}`, in this case doubling `num`:
+
+```crystal
+@[ARTA::Get(path: "/multiply/{num}")]
+def multiply(@[Multiply(by: 2)] num : Int32) : Int32
+  num
+end
```

Comment:
I don't think this is really a good example. It's unnecessarily complex for illustrating the feature. Alone the fact that there are actually two annotations makes it non-trivial to grok. And it's by far not self-contained. In order to understand what  the code really means, you need some context about the Athena router.

Instead we could perhaps use the example from https://github.com/crystal-lang/crystal/issues/12039#issuecomment-1118859028:
```cr
# This annotation designates that a method parameter could be unused.
# It may be used to suppress a respective linter message.
annotation MaybeUnused; end

def foo(@[MaybeUnused] x)
end 
```
This is still made up and not currently implemented in Ameba or anything, but I think it's better than the router example.

I'm happy for an even better, minimal example.